src/libraries/Common/src/Interop/Windows/NCrypt/Interop.Properties.cs

```diff
@@ -20,62 +20,43 @@ internal static partial class NCrypt
         internal static extern unsafe ErrorCode NCryptSetProperty(SafeNCryptHandle hObject, string pszProperty, [In] void* pbInput, int cbInput, CngPropertyOptions dwFlags);
 
         [SupportedOSPlatform("windows")]
-        internal static ErrorCode NCryptGetByteProperty(SafeNCryptHandle hObject, string pszProperty, ref byte result, CngPropertyOptions options = CngPropertyOptions.None)
+        internal static unsafe ErrorCode NCryptGetByteProperty(SafeNCryptHandle hObject, string pszProperty, ref byte result, CngPropertyOptions options = CngPropertyOptions.None)
         {
-            int cbResult;
-            ErrorCode errorCode;
-
-            unsafe
+            fixed (byte* pResult = &result)
             {
-                fixed (byte* pResult = &result)
-                {
-                    errorCode = Interop.NCrypt.NCryptGetProperty(
-                        hObject,
-                        pszProperty,
-                        pResult,
-                        sizeof(byte),
-                        out cbResult,
-                        options);
-                }
+                ErrorCode errorCode = Interop.NCrypt.NCryptGetProperty(
+                    hObject,
+                    pszProperty,
+                    pResult,
+                    sizeof(byte),
+                    out int cbResult,
+                    options);
+
+                Debug.Assert(errorCode != ErrorCode.ERROR_SUCCESS || cbResult == sizeof(byte));
```

Comment:
Nit: I find the earlier pattern (`if (success) { assert( /* ... */ ); }`) easier to read.

(Same comment applies elsewhere in this file.)

Reply:
Reverted